### ISSUE TITLE
Validation error with Depth buffers

### ISSUE THREAD

**Issue by cloudhead** [gfx-rs/wgpu#384]
I'm getting the following validation error when using depth writing/testing in certain cases:
```
[2019-11-26T22:42:24Z ERROR gfx_backend_vulkan]
VALIDATION [VUID-VkImageSubresource-aspectMask-parameter (0)] : vkCmdPipelineBarrier(): Depth-only image formats must have the VK_IMAGE_ASPECT_DEPTH_BIT set. The Vulkan spec states: aspectMask must be a valid combination of VkImageAspectFlagBits values (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-VkImageSubresource-aspectMask-parameter)
object info: (type: IMAGE, hndl: 116)
```

I don't know if I'm doing anything wrong, since things work as expected.. the relevant part of my pipeline config looks like this:

```rust
            .create_render_pipeline(&wgpu::RenderPipelineDescriptor {
             ...
                rasterization_state: Some(wgpu::RasterizationStateDescriptor {
                    front_face: wgpu::FrontFace::Ccw,
                    cull_mode: wgpu::CullMode::None,
                    depth_bias: 0,
                    depth_bias_slope_scale: 0.0,
                    depth_bias_clamp: 0.0,
                }),
...
                depth_stencil_state: Some(wgpu::DepthStencilStateDescriptor {
                    format: wgpu::TextureFormat::Depth32Float,
                    depth_write_enabled: true,
                    depth_compare: wgpu::CompareFunction::LessEqual,
                    stencil_front: wgpu::StencilStateFaceDescriptor::IGNORE,
                    stencil_back: wgpu::StencilStateFaceDescriptor::IGNORE,
                    stencil_read_mask: 0,
                    stencil_write_mask: 0,
                }),
                index_format: wgpu::IndexFormat::Uint16,
                vertex_buffers: &[vertex_attrs],
                sample_count: 1,
                sample_mask: !0,
                alpha_to_coverage_enabled: false,
            });
```

**Comment by cloudhead** [comment]
The depth textures only have
```
usage: wgpu::TextureUsage::COPY_DST | wgpu::TextureUsage::OUTPUT_ATTACHMENT
```

**Comment by kvark** [comment]
Thank you for the report! Is this on wgpu-0.4 or master?
It would be useful to know exactly, which line of code triggers this.

**Comment by cloudhead** [comment]
This is wgpu 0.4. I’ll try and narrow it down to the line of code! I could also try master if the API hasn’t changed.

**Comment by cloudhead** [comment]
So the offending line is:
```
        encoder.copy_buffer_to_texture(
            wgpu::BufferCopyView {
                buffer,
                offset: 0,
                row_pitch: 4 * w,
                image_height: h,
            },
            wgpu::TextureCopyView {
                texture,
                mip_level: 0,
                array_layer: 0,
                origin: wgpu::Origin3d { x, y, z: 0.0 },
            },
            extent,
        );
```
On a depth buffer.

I am creating the buffer and copying to it in the same frame, so this might be the issue, though I do this for color textures without problem.

**Comment by kvark** [comment]
I can see that the code would use the full aspects of the texture, so in your case it should be correct: https://github.com/gfx-rs/wgpu/blob/469c086a1ae8843f87296f0b29dd7a6031aca3f9/wgpu-core/src/command/transfer.rs#L140

Is there any way I could reproduce this? Perhaps, you could share a branch of code?

**Comment by cloudhead** [comment]
Ok, I reproduced it pretty easily in `rgx`, here's the branch: https://github.com/cloudhead/rgx/tree/bug/depth-buffer-validation-error

You have to run `cargo run --example screenshot`

**Comment by cloudhead** [comment]
You can grep for `XXX` for the offending line(s)

**Comment by kvark** [comment]
Turns out, this was already fixed in #378, just never backported to 0.4 version. Doing this now in #389

**Comment by cloudhead** [comment]
:pray:

**Comment by kvark** [comment]
Fixed by #378 and #389
